client/src/main/java/io/pravega/client/netty/impl/ConnectionFactoryImpl.java

```diff
@@ -49,6 +53,9 @@
     private EventLoopGroup group;
     private boolean nio = false;
     private final AtomicBoolean closed = new AtomicBoolean(false);
+    private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(
+            Runtime.getRuntime().availableProcessors(),
```

Comment:
Filed issue #1521 for this. Will take it up as part of this issue. The thread pool which will be used by writers and readers should be configurable.